sklearn/metrics/_classification.py

```diff
@@ -1849,7 +1850,9 @@ def precision_recall_fscore_support(
         true_sum = xp.reshape(xp.sum(true_sum), (1,))
 
     # Finally, we have all our sufficient statistics. Divide! #
-    beta2 = beta**2
+    device_ = device(y_true, y_pred)
```

Comment:
In line 1846, we could use 
`xp, _, device = get_namespace_and_device(y_true, y_pred)`
to eliminate this line. Just my two cents!